rules/do-you-optimize-tinacms-project/rule.md

```diff
@@ -0,0 +1,103 @@
+
+<!--endintro-->
+
+## 1. Structuring your TinaCMS Architecture
+
+When working with large datasets or generating multiple subcomponents, following best practices is crucial to maintain performance and clarity.
+
+### ❌ Bad practices
+
+* **Making individual requests with Tina client for each Subcomponent**
+
+  * This method can overwhelm the build process, particularly when dealing with a large number of subcomponents.
+* **Using deeply nested schemas with nested references**
+
+  * Complex and deeply nested schemas increase the complexity of the project, making it harder to manage and more prone to build failures.
+  * They can also lead to inefficient data fetching, further slowing down both runtime and build processes.
+
+### ✅ Good practices
+
+* **Making a single request at a top-level server component and passing props down**
+
+  * Data fetched at the top level is cached by default, which enhances performance by reducing redundant API calls.
+  * Subcomponents can directly access the necessary data, eliminating the need for repeated requests.
+
+::: good
+![Figure: Good example - Single request at the top-level server and passing props down](2024-08-28_16-21-56.png)
+:::
+
+* **Caching data at a Top-level and accessing it when necessary**
+
+  * If passing props is not feasible (e.g., when a component depends on Next.js router information), you should make a general top-level request, cache the data, and then access it directly from the cache within the component.
+  * This approach ensures efficient data retrieval and reduces the server load at build time.
+
+## 2. Improving Runtime Performance
+
+Optimizing runtime performance is key to delivering a fast and responsive user experience.
+
+### ❌ Bad practices
+
+* **Using client-side requests instead of cached values from server-side components**
+
+  * This approach bypasses pre-fetched, cached data, increasing server load and slowing down the application.
+  * It negates the benefits of static site generation, where data should be ready and waiting.
+
+### Common practices
+
+* **Utilizing Next.js caching via fetch** 
+
+  * Next.js’s fetch API is designed to cache by default during static generation and server-side rendering.
+
+### ✅ Good practices
+
+* **Utilizing caching at the highest level of the application**
+
+  * By caching data at the top level, you ensure that it is readily available throughout your application, reducing redundant data fetching and improving overall performance.
+  * This approach allows for more efficient data management and quicker response times.
+* **Using dedicated caching packages**
+
+  * [**Redis**:](https://redis.io/fr/solutions/cas-dutilisation/cache/) An in-memory data structure store, Redis is fast and versatile, making it ideal for caching data in memory. 
+  * [**Node-cache**:](https://www.npmjs.com/package/node-cache) A lightweight, in-memory caching solution for Node.js applications.
+
```

Comment:
still unsure about this section. I feel we should mention to statically generate the content i.e. calling it only on build. Or using a fetching library like react query that can cache frequent requests.

Reply:
agree - it also focuses on nextjs and react as opposed to TinaCMS

Reply:
I changed this part, more focus on Tina, not on next/react.